sql/opt/exec/explain: TestMaxDiskSpillUsage failed

sql/opt/exec/explain.TestMaxDiskSpillUsage [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13130027?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13130027?buildTab=artifacts#/) on release-23.2.0-rc @ [ca2c36f76f16e664addc43f32f2626a366969b4f](https://github.com/cockroachdb/cockroach/commits/ca2c36f76f16e664addc43f32f2626a366969b4f):

Fatal error:

```
panic: test timed out after 55s
running tests:
	TestMaxDiskSpillUsage (55s)

```
Stack: 

```
goroutine 6265 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x320
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestMaxDiskSpillUsage
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/3286699bf278cb16f63aa79f134a388f/logTestMaxDiskSpillUsage4189841478
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:159: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    output_test.go:163: -- test log scope end --
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMaxDiskSpillUsage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34611

--- a/pkg/sql/opt/exec/explain/BUILD.bazel
+++ b/pkg/sql/opt/exec/explain/BUILD.bazel
@@ -66,7 +66,6 @@ go_test(
         "//pkg/security/securitytest",
         "//pkg/server",
         "//pkg/sql/catalog/colinfo",
-        "//pkg/sql/execinfra",
         "//pkg/sql/opt/cat",
         "//pkg/sql/opt/exec",
         "//pkg/sql/opt/memo",

--- a/pkg/sql/opt/exec/explain/output_test.go
+++ b/pkg/sql/opt/exec/explain/output_test.go
@@ -21,7 +21,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/multitenant/tenantcapabilities"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
-	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
 	"github.com/cockroachdb/cockroach/pkg/sql/opt/exec/explain"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
@@ -124,24 +123,14 @@ func TestMaxDiskSpillUsage(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	testClusterArgs := base.TestClusterArgs{
-		ReplicationMode: base.ReplicationAuto,
-	}
-	distSQLKnobs := &execinfra.TestingKnobs{}
-	distSQLKnobs.ForceDiskSpill = true
-	testClusterArgs.ServerArgs.Knobs.DistSQL = distSQLKnobs
-	tc := testcluster.StartTestCluster(t, 1, testClusterArgs)
 	ctx := context.Background()
-	defer tc.Stopper().Stop(ctx)
+	s, conn, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	defer s.Stopper().Stop(ctx)
 
-	conn := tc.Conns[0]
+	sqlDB := sqlutils.MakeSQLRunner(conn)
+	sqlDB.Exec(t, "CREATE TABLE t (a PRIMARY KEY, b) AS SELECT i, i FROM generate_series(1, 10) AS g(i)")
 
-	_, err := conn.ExecContext(ctx, `
-CREATE TABLE t (a PRIMARY KEY, b) AS SELECT i, i FROM generate_series(1, 10) AS g(i)
-`)
-	assert.NoError(t, err)
 	maxDiskUsageRE := regexp.MustCompile(`max sql temp disk usage: (\d+)`)
-
 	queryMatchRE := func(query string, re *regexp.Regexp) bool {
 		rows, err := conn.QueryContext(ctx, query)
 		assert.NoError(t, err)
@@ -155,7 +144,8 @@ CREATE TABLE t (a PRIMARY KEY, b) AS SELECT i, i FROM generate_series(1, 10) AS 
 		return false
 	}
 
-	// We are expecting disk spilling to show up because we enabled ForceDiskSpill
+	// Use very low workmem limit so that the disk spilling happens.
+	sqlDB.Exec(t, "SET distsql_workmem = '2B';")
 	// knob above.
 	assert.True(t, queryMatchRE(`EXPLAIN ANALYZE (VERBOSE, DISTSQL) select * from t join t AS x on t.b=x.a`, maxDiskUsageRE), "didn't find max sql temp disk usage: in explain")
 	assert.False(t, queryMatchRE(`EXPLAIN ANALYZE (VERBOSE, DISTSQL) select * from t `, maxDiskUsageRE), "found unexpected max sql temp disk usage: in explain")
@@ -173,15 +163,8 @@ func TestCPUTimeEndToEnd(t *testing.T) {
 		return
 	}
 
-	testClusterArgs := base.TestClusterArgs{
-		ReplicationMode: base.ReplicationAuto,
-	}
-	distSQLKnobs := &execinfra.TestingKnobs{}
-	distSQLKnobs.ForceDiskSpill = true
-	testClusterArgs.ServerArgs.Knobs.DistSQL = distSQLKnobs
 	const numNodes = 3
-
-	tc := testcluster.StartTestCluster(t, numNodes, testClusterArgs)
+	tc := testcluster.StartTestCluster(t, numNodes, base.TestClusterArgs{})
 	ctx := context.Background()
 	defer tc.Stopper().Stop(ctx)
 

